XRTK-Core/Packages/com.xrtk.core/Runtime/Providers/Controllers/Hands/HandDataPostProcessor.cs

```diff
@@ -0,0 +1,230 @@
+            UpdateIsGrippingAndStrength(handData);
+            UpdatePointerPose(handData);
+            UpdateTrackedPose(handData);
+        }
+
+        /// <summary>
+        /// Updates <see cref="HandData.IsGripping"/> and <see cref="HandData.GripStrength"/>
+        /// if the platform did not provide it.
+        /// </summary>
+        /// <param name="handData">The hand data to update <see cref="HandData.IsGripping"/> and <see cref="HandData.GripStrength"/> for.</param>
+        private void UpdateIsGrippingAndStrength(HandData handData)
+        {
+            if (handData.IsTracked)
+            {
+                var palmPose = handData.Joints[(int)TrackedHandJoint.Palm];
+                var pinkyTipPose = handData.Joints[(int)TrackedHandJoint.PinkyTip];
+                var ringTipPose = handData.Joints[(int)TrackedHandJoint.RingTip];
+                var middleTipPose = handData.Joints[(int)TrackedHandJoint.MiddleTip];
+                var indexTipPose = handData.Joints[(int)TrackedHandJoint.IndexTip];
+
+                if (!PlatformProvidesIsGripping)
+                {
+                    handData.IsGripping = (palmPose.Position - pinkyTipPose.Position).sqrMagnitude <= TEN_CENTIMETER_SQUARE_MAGNITUDE &&
+                    (palmPose.Position - ringTipPose.Position).sqrMagnitude <= TEN_CENTIMETER_SQUARE_MAGNITUDE &&
+                    (palmPose.Position - middleTipPose.Position).sqrMagnitude <= TEN_CENTIMETER_SQUARE_MAGNITUDE &&
+                    (palmPose.Position - indexTipPose.Position).sqrMagnitude <= TEN_CENTIMETER_SQUARE_MAGNITUDE;
+                }
+
+                if (!PlatformProvidesGripStrength)
+                {
+                    var averageDistanceSquareMagnitude = ((palmPose.Position - pinkyTipPose.Position).sqrMagnitude - TEN_CENTIMETER_SQUARE_MAGNITUDE +
+                        (palmPose.Position - ringTipPose.Position).sqrMagnitude - TEN_CENTIMETER_SQUARE_MAGNITUDE +
+                        (palmPose.Position - middleTipPose.Position).sqrMagnitude - TEN_CENTIMETER_SQUARE_MAGNITUDE +
+                        (palmPose.Position - indexTipPose.Position).sqrMagnitude - TEN_CENTIMETER_SQUARE_MAGNITUDE) / 4f;
+                    handData.GripStrength = 1 - Mathf.Clamp(averageDistanceSquareMagnitude / GRIP_STRENGTH_DISTANCE, 0f, 1f);
+
+                    Debug.Log(handData.GripStrength);
+                }
+            }
+            else
+            {
+                handData.IsGripping = false;
+                handData.GripStrength = 0f;
+            }
+        }
+
+        /// <summary>
+        /// Updates <see cref="HandData.IsPinching"/> and <see cref="HandData.PinchStrength"/>
+        /// if the platform did not provide it.
+        /// </summary>
+        /// <param name="handData">The hand data to update <see cref="HandData.IsPinching"/> and <see cref="HandData.PinchStrength"/> for.</param>
+        private void UpdateIsPinchingAndStrength(HandData handData)
```

Comment:
Would be nice to expose pinch strength for different fingers

Reply:
What would this be useful for?

Reply:
Thanks Eric. It's on my list for the next iteration. I'll evaluate use cases for it. I already have curl strength for each finger as of now.